todo/src/reducer/reducer.js

```diff
@@ -0,0 +1,47 @@
+import { ADD_TODOS, TOGGLE_TODOS, DELETE_TODOS } from '../action';
+
+const initialState = {
+    todos : [
+        {
+            value: 'Walk the dog.',
+            completed: false
+        }
+    ]
+}
+
+function reducer( state=initialState, action ) {
+    switch(action.type) {
+        case ADD_TODOS:
+        return {
+            ...state,
+            todos: [...state.todos, {
+                value: action.payload,
+                completed: false
+                }]
+        }
+        
+        case TOGGLE_TODOS:
+        return {
+            ...state,
+            todos: state.todos.map((todo, index) => {
+                if(index === action.payload)
+                    return{...todo, completed: !todo.completed}
+                
+                return todo;
+            })
+        } 
+
+        // case DELETE_TODOS:
```

Comment:
This is on the right track! Great idea to use the filter method. Just remember the function inside filter expects you to return true or false, and that an element will only be added to the new array if you return true inside of the filter method. Looking awesome so far!